app/models/tag.rb

```diff
@@ -0,0 +1,19 @@
+class Tag < ApplicationRecord
+  belongs_to :job
+  has_many :job_tags
+  has_many :jobs, through: :job_tags
+    validates :text, presence: true
+
+  def count_of_jobs_with_tag
+    JobTag
+    .where(tag_id: self.id)
+    .count
+  end
+
+  def average_salary_for_jobs_with_tag
+    Tag
+    .joins(:job_tags)
+    .joins(:jobs)
+    .average(:salary)
+  end
```

Comment:
I actually think this is off. Believe it will return an average for all jobs regardless of whether they have a particular tag or not. In your test would need to create a tag that had two jobs and another tag that only had one of those two jobs and check the average for both.

Similarly, here we can use our existing relationships. `jobs` will find all of the jobs associated with a particular tag, 

```
jobs.average(:salary)
```

Would like to see a model test for both of these methods.